src/context/product-context.js

```diff
@@ -1,20 +1,21 @@
 import { createContext, useContext, useReducer } from "react";
-import { productReducer } from "../reducers/productReducer";
+import { productReducer, initialProducts } from "../reducers/productReducer";
 
 const ProductContext = createContext();
-const useProduct = () => {
-  return useContext(ProductContext);
-};
-
 const ProductProvider = ({ children }) => {
-  const [productState, productDispatch] = useReducer(productReducer, {
-    data: [],
-  });
+  const [productState, productDispatch] = useReducer(
+    productReducer,
+    initialProducts
+  );
   return (
     <ProductContext.Provider value={{ productState, productDispatch }}>
       {children}
     </ProductContext.Provider>
   );
 };
 
+const useProduct = () => {
+  return useContext(ProductContext);
```

Comment:
This can be written as const useProduct = ()=> useContext(ProductContext) but if you prefer it this way works fine too.

Reply:
Thanks for the suggestion